packages/Search/test/tstBoostGeometryAdapters.cpp

```diff
@@ -0,0 +1,249 @@
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <DTK_DetailsAlgorithms.hpp>
+
+#include <Teuchos_UnitTestHarness.hpp>
+
+#include "DTK_BoostGeometryAdapters.hpp"
+
+namespace bg = boost::geometry;
+namespace dtk = DataTransferKit::Details;
+// Conveniently importing Point and Box in DataTransferKit::Details:: namespace
+// and declaring type aliases within boost::geometry:: so that we are able to
+// just use dtk:: and bg:: to specify what geometry or algorithm we mean.
+namespace DataTransferKit
+{
+namespace Details
+{
+using DataTransferKit::Box;
+using DataTransferKit::Point;
+}
+}
+namespace boost
+{
+namespace geometry
+{
+using Point = model::point<double, 3, cs::cartesian>;
+using Box = model::box<Point>;
+}
+}
+
+TEUCHOS_UNIT_TEST( BoostGeometryAdapters, equals )
+{
+    dtk::Point point = {{1., 2., 3.}};
+    TEST_ASSERT( dtk::equals( point, {{1., 2., 3.}} ) );
+    TEST_ASSERT( !dtk::equals( point, {{0., 0., 0.}} ) );
+    TEST_ASSERT( bg::equals( point, bg::make<dtk::Point>( 1., 2., 3. ) ) );
+    TEST_ASSERT( !bg::equals( point, bg::make<dtk::Point>( 4., 5., 6. ) ) );
+    TEST_ASSERT( bg::equals( point, bg::make<bg::Point>( 1., 2., 3. ) ) );
+    TEST_ASSERT( !bg::equals( point, bg::make<bg::Point>( 0., 0., 0. ) ) );
+
+    dtk::Box box = {{{1., 2., 3.}}, {{4., 5., 6.}}};
+    TEST_ASSERT( dtk::equals( box, {{{1., 2., 3.}}, {{4., 5., 6.}}} ) );
+    TEST_ASSERT( !dtk::equals( box, {{{0., 0., 0.}}, {{1., 1., 1.}}} ) );
+    TEST_ASSERT( bg::equals( box, dtk::Box{{{1., 2., 3.}}, {{4., 5., 6.}}} ) );
+    TEST_ASSERT( !bg::equals( box, dtk::Box{{{0., 0., 0.}}, {{1., 1., 1.}}} ) );
+    TEST_ASSERT(
+        bg::equals( box, bg::Box( bg::make<bg::Point>( 1., 2., 3. ),
+                                  bg::make<bg::Point>( 4., 5., 6. ) ) ) );
+    TEST_ASSERT(
+        !bg::equals( box, bg::Box( bg::make<bg::Point>( 0., 0., 0. ),
+                                   bg::make<bg::Point>( 1., 1., 1. ) ) ) );
+    // Now just for fun compare the DTK box to a Boost.Geometry box composed of
+    // DTK points.
+    TEST_ASSERT( bg::equals(
+        box, bg::model::box<dtk::Point>( {{1., 2., 3.}}, {{4., 5., 6.}} ) ) );
+    TEST_ASSERT( !bg::equals(
+        box, bg::model::box<dtk::Point>( {{0., 0., 0.}}, {{0., 0., 0.}} ) ) );
+}
+
+TEUCHOS_UNIT_TEST( BoostGeometryAdapters, distance )
+{
+    dtk::Point a = {{0., 0., 0.}};
+    dtk::Point b = {{0., 1., 0.}};
+    TEST_EQUALITY( dtk::distance( a, b ), 1. );
+    TEST_EQUALITY( bg::distance( a, b ), 1. );
+
+    std::tie( a, b ) = std::make_pair<dtk::Point, dtk::Point>( {{0., 0., 0.}},
+                                                               {{1., 1., 1.}} );
+    TEST_EQUALITY( dtk::distance( a, b ), std::sqrt( 3. ) );
```

Comment:
I am surprised this work with the round-off...

Reply:
Well DTK distance does call `std::sqrt()` so if anything I was more expecting it to fail with Boost.
I will move the NOTE about testing for floating point equality here.  I think it is fine to keep it the way for now.